src/xercesc/util/Janitor.hpp

```diff
@@ -155,8 +155,10 @@ private :
 };
 
 
+#if defined(__GNUC__) || (! defined(_AIX) && ! defined(__hpux) && ! defined(__sun))
```

Comment:
Is the `defined(__GNUC__)` necessary?  Isn't that already taken care of by the other part of the or?

Reply:
The __GNUC__ is necessary because we're talking here about native vendor's compiler (e.g. not gcc on AIX/Solaris). GCC on AIX/Solaris will be OK even without this change.